### PR TITLE
make log dir configurable and make log name compatible with RocketMQ main project

### PR DESCRIPTION
**Pull request by humkum**
## What is the purpose of the change

#400

## Brief changelog

Add the log path to the environment variable by using setenv function, and set this path before the log object is initialized.

## Verifying this change

By setting producer log path to "/logs/producer/cpp/", the log file was saved in "~/logs/producer/cpp/", and the main log file name is "rocketmq_client.log", backup log file name's format is "rocketmq_client_%Y%m%d-%N.log". As follows:

image

## MoreOver

We use boost1.78.0 to make backup log file name configurable.

Follow this checklist to help us incorporate your contribution quickly and easily. Notice, `it would be helpful if you could finish the following 5 checklist(the last one is not necessary)before request the community to review your PR`.

- [x] Make sure there is a [Github issue](https://github.com/apache/rocketmq/issues) filed for the change (usually before you start working on it). Trivial changes like typos do not require a Github issue. Your pull request should address just this issue, without pulling in other changes - one PR resolves one issue. 
- [x] Format the pull request title like `[ISSUE #123] Fix UnknownException when host config not exist`. Each commit in the pull request should have a meaningful subject line and body.
- [x] Write a pull request description that is detailed enough to understand what the pull request does, how, and why.
- [x] Write necessary unit-test(over 80% coverage) to verify your logic correction, more mock a little better when a cross-module dependency exists.
- [ ] If this contribution is large, please file an [Apache Individual Contributor License Agreement](http://www.apache.org/licenses/#clas).

### CODE REVIEW

File: src/log/Logging.cpp [hunk 1]
```diff
@@ -38,16 +38,19 @@ logAdapter* logAdapter::getLogInstance() {
...
+  std::string fileName = "rocketmq_client.log";
   m_logFile += fileName;
 
   // boost::log::expressions::attr<
   // boost::log::attributes::current_thread_id::value_type>("ThreadID");
   boost::log::register_simple_formatter_factory<boost::log::trivial::severity_level, char>("Severity");
-  m_logSink = logging::add_file_log(keywords::file_name = m_logFile, keywords::rotation_size = 100 * 1024 * 1024,
+  m_logSink = logging::add_file_log(keywords::file_name = m_logFile,
+                                    keywords::target_file_name = "rocketmq_client_%Y%m%d-%N.log",
+                                    keywords::rotation_size = 10 * 1024,
```
Comment (ifplusor) [inline src/log/Logging.cpp:53]: These are breaking changes and should be considered carefully. I suggest splitting these into a separate PR.

**Review by ifplusor — CHANGES_REQUESTED** [review]

File: src/log/Logging.cpp (hunk 1 above)
Comment (humkum) [reply src/log/Logging.cpp]: Here we use boost 1.78.0 to make the backup log file name configurable. Did you mean split this change into another PR?

Comment (ifplusor) [reply src/log/Logging.cpp]: Make log name compatible with RocketMQ main project, and rotation size.

Comment (humkum) [reply src/log/Logging.cpp]: OK， I will put modification of rotation size in another PR.

File: src/log/Logging.cpp [hunk 2]
```diff
@@ -99,10 +102,19 @@ elogLevel logAdapter::getLogLevel() {
   return m_logLevel;
 }
 
+void logAdapter::setLogDir() {
+  char* p;
+  if((p = getenv(ROCKETMQ_CLIENT_LOG_DIR.c_str()))) {
+    m_log_dir = p;
```
[the remark this answers is not in the document]
Comment (humkum) [reply src/log/Logging.cpp:115]: Done~

File: src/log/Logging.cpp [hunk 3]
```diff
@@ -99,9 +102,25 @@ elogLevel logAdapter::getLogLevel() {
   return m_logLevel;
 }
 
+void logAdapter::setLogDir() {
+  char* p = new char[10];
+  if((p = getenv(ROCKETMQ_CLIENT_LOG_DIR.c_str()))) {
+    int len = strlen(p);
+    if (p[len - 1] != '/') {
+      char* log_dir = new char[len + 1];
```
Comment (ifplusor) [inline src/log/Logging.cpp:110]: `len + 2` for '/' and '\0', and no `delete` for `new`

File: src/log/Logging.cpp [hunk 4]
```diff
@@ -99,9 +102,25 @@ elogLevel logAdapter::getLogLevel() {
   return m_logLevel;
 }
 
+void logAdapter::setLogDir() {
+  char* p = new char[10];
```
Comment (ifplusor) [inline src/log/Logging.cpp:106]: This `new` is unnecessary.

File: src/log/Logging.cpp [hunk 5]
```diff
@@ -99,9 +102,25 @@ elogLevel logAdapter::getLogLevel() {
...
+      char* log_dir = new char[len + 1];
+      strcpy(log_dir, p);
+      strcat(log_dir, "/");
+      p = log_dir;
+    }
+    m_log_dir = p;
+  } else {
+    m_log_dir = "/logs/rocketmq-client/";
+  }
+}
```
Comment (ifplusor) [inline src/log/Logging.cpp:119]:
```c++
void logAdapter::setLogDir() {
  char* p = nullptr;
  if (p = getenv(ROCKETMQ_CLIENT_LOG_DIR.c_str())) {
    m_log_dir = p;
    if (m_log_dir[m_log_dir.length() - 1] != '/') {
      m_log_dir += '/';
    }
  } else {
    m_log_dir = "/logs/rocketmq-client/";
  }
}
```

**Review by ifplusor — CHANGES_REQUESTED** [review]

File: src/log/Logging.cpp (hunk 5 above)
Comment (ifplusor) [reply src/log/Logging.cpp:119]: and do trim on m_log_dir

**Review by ifplusor — APPROVED** [review]
LGTM

**Review by ShannonDing — APPROVED** [review]

**Pull request merged**